app/assets/javascripts/discourse/app/components/search-menu/results/assistant-item.js

```diff
@@ -95,7 +95,10 @@ export default class AssistantItem extends Component {
   @debounce(100)
   itemSelected() {
     let updatedTerm = "";
-    if (this.args.slug && this.args.suggestionKeyword) {
+    if (
+      this.args.slug &&
+      (this.args.suggestionKeyword || this.args.concatSlug)
+    ) {
```

Comment:
A downside of the `AssistantItem` being reusable is that we have to account for many different use cases. Adding a `concatSlug` argument allows us to have better control over when we do or do not want to populate the slug (of the clicked assistant item) to the search input.